design_docs/rsyslog_new_input_output_roles_proposal.md

```diff
@@ -0,0 +1,240 @@
+# Rsyslog - Proposal for New Input and Output Role Configuration
+
+## Background
+
+The logging role defines `rsyslog_default` variable to deploy the original, all-in-one rsyslog configuration file rsyslog.conf.
+It configures to get inputs from `imjournal` and output to the local files in /var/log.
+It was originally added to provide "noop" to the logging role.
+Then, it was utilized to support for the local system logs to store into the local files combining with the openshift logging in the ovirt logging.
+In the linux-system-roles review process, some critical questions on `rsyslog_default` were raised, such as the original intention - providing "noop" is not the style ansible recommends and no other linux-system-roles do not nor will not have such a default variable.
+Considering the inputs, the functionality that `rsyslog_default` offers has to be achieved by the logging role itself instead of copying the hardcoded rsyslog.conf to the target host.
+
+This proposal will discuss about these items:
+- how to replace the `rsyslog_default` variable with the logging role input/output role configuration.
+- how to support more generic way to connect input roles and output roles.
+
+Please note that this github commit starting this doc implements the first cut of the proposal including:
+- input roles - basics input role (`imjournal`, `imuxsock`, `imudp`, `imptcp`) and files input role (`imfile`), and
+- output roles - files output role (`omfile`), forwards output role (`omfwd`) and elasticsearch role (`omelasticsearch`).
+Leftover tasks, other roles to be changed and eliminating `rsyslog_default`, are discussed in [To Do's](#to-dos).
+
+## Proposals
+
+### Output roles
+Each output action is in the ruleset which name is the logging_outputs name.
+
+Elasticsearch, Files and Forwards output configuration files in /etc/rsyslog.d:
+```
+  10-output-modules.conf
+  30-output-elasticsearch.conf
+  30-output-files.conf
+  30-output-forwards.conf
+```
+With configuring the outputs (in this example, `files`) with a unique name (e.g., "files_output1", etc.), it will generate an rsyslog configuraion in the ruleset which name is the unique name.
```

Comment:
what is the example variable structure to generate this?
In the exam,ples above, you have `name: unique_output_name`, but here the name is `files_output1`. Can you please make the examples consistent? I.e. if you show an example of variables, show later what would be the configuration produced by that very same example?

Reply:
Ok.  Let me put an example first and explain based on it.

Reply:
Done.